core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/pagination/PaginationSP.java

```diff
@@ -46,66 +46,80 @@ public String printCounter() {
       }
 
+      result.append("<div class=\"pageClicToGo\">");
+
       // display previous link (or nothing if current page is first one)
       if (getFirstItemIndex() >= getNbItemsPerPage()) {
-
         if (currentPage - getNumberOfPagesAround() > 1) {
           // display first page link
-          result.append("<div class=\"pageOff\">");
+          result.append("<a class=\"pageOff\"");
           result.append(getLink(javascriptFunc, 0, getString("GEF.pagination.firstPage")));
           result.append("<img src=\"").append(getIconsPath()).append(
-              "/arrows/arrowDoubleLeft.gif\" border=\"0\" align=\"absmiddle\" alt=\"")
-              .append(getString("GEF.pagination.firstPage")).append("\"/></a>");
-          result.append("</div>");
+              "/arrows/arrowDoubleLeft.png\" border=\"0\" align=\"absmiddle\" alt=\"")
+              .append(getString("GEF.pagination.firstPage")).append("\"/>");
+          result.append("</a>");
         }
 
         // display previous page link
-        result.append("<div class=\"pageOff\">");
+        result.append("<a class=\"pageOff\"");
         int index = getIndexForPreviousPage();
         result.append(getLink(javascriptFunc, index, getString("GEF.pagination.previousPage")));
         result.append("<img src=\"").append(getIconsPath()).append(
-            "/arrows/arrowLeft.gif\" border=\"0\" align=\"absmiddle\" alt=\"")
-            .append(getString("GEF.pagination.previousPage")).append("\"/></a>");
-        result.append("</div>");
-
+            "/arrows/arrowLeft.png\" border=\"0\" align=\"absmiddle\" alt=\"")
+            .append(getString("GEF.pagination.previousPage")).append("\"/>");
+        result.append("</a>");
       }
 
       // display all pages
-      for (int i = 1; i <= nbPage; i++) {
-        if (i == currentPage) {
-          result.append("<div class=\"pageOn\">");
-          result.append(i).append("</div>");
-        } else {
-          int nbDisplayPages = getNumberOfPagesAround();
-          // display 10 pages (or less) before current page
-          // display 10 pages (or less) after current page
-          if (currentPage - nbDisplayPages <= i && i <= currentPage + nbDisplayPages) {
-            int index = getIndexForDirectPage(i);
-            result.append("<div class=\"pageOff\">");
-            result.append(getLink(javascriptFunc, index, getString("GEF.pagination.gotoPage") + i));
+      if (nbPage > 1) {
+        for (int i = 1; i <= nbPage; i++) {
+          if (i == currentPage) {
+            result.append("<a class=\"pageOn\">");
             result.append(i).append("</a>");
-            result.append("</div>");
+          } else {
+            int nbDisplayPages = getNumberOfPagesAround();
+            // display 10 pages (or less) before current page
+            // display 10 pages (or less) after current page
+            if (currentPage - nbDisplayPages <= i && i <= currentPage + nbDisplayPages) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS134)